### PR TITLE
[PT][NEURON][BUILD] - Add PT1.9 for neuron

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Added PT1.9 for neuron. Also this is using the latest release where there is no neuron runtime. Made the corresponding changes

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+ENV LD_LIBRARY_PATH /opt/conda/lib/:$LD_LIBRARY_PATH
+ENV PATH /opt/conda/bin:$PATH
+ENV SAGEMAKER_SERVING_MODULE sagemaker_pytorch_serving_container.serving:main
+ENV TEMP=/home/model-server/tmp
+
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends software-properties-common \
+    && add-apt-repository ppa:openjdk-r/ppa \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron:22]: nit: Please correct the indentation here for readability:
```suggestion
 && apt-get install -y --no-install-recommends software-properties-common \
 && add-apt-repository ppa:openjdk-r/ppa \
 && apt-get update \
 && apt-get install -y --no-install-recommends \
```

File: pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    openjdk-11-jdk \
+    vim \
+    wget \
+    unzip \
+    zlib1g-dev \
+    libcap-dev \
+    gpg-agent \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /tmp/tmp* \
+    && apt-get clean
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron:45]: nit: Please correct the indentation here for readability:
```suggestion
 && rm -rf /var/lib/apt/lists/* \
 && rm -rf /tmp/tmp* \
 && apt-get clean
```

File: pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+RUN echo "deb https://apt.repos.neuron.amazonaws.com bionic main" > /etc/apt/sources.list.d/neuron.list
+RUN wget -qO - https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEURON.PUB | apt-key add -
+
+RUN apt-get update \
+    && apt-get install -y \
+    aws-neuron-tools \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /tmp/tmp* \
+    && apt-get clean
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron:55]: nit: Please correct the indentation here for readability:
```suggestion
RUN apt-get update \
 && apt-get install -y \
    aws-neuron-tools \
 && rm -rf /var/lib/apt/lists/* \
 && rm -rf /tmp/tmp* \
 && apt-get clean
```

File: pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+ && chown -R model-server /home/model-server /opt/ml/model
+
+COPY neuron-entrypoint.py /usr/local/bin/dockerd-entrypoint.py
+COPY neuron-start.sh /usr/local/bin/neuron-start.sh
+COPY torchserve-neuron.sh /usr/local/bin/entrypoint.sh
+COPY config.properties /home/model-server
+
+RUN chmod +x /usr/local/bin/dockerd-entrypoint.py
+RUN chmod +x /usr/local/bin/neuron-start.sh
+RUN chmod +x /usr/local/bin/entrypoint.sh
```
Comment (saimidu) [inline pytorch/inference/docker/1.9/py3/sdk1.16.0/Dockerfile.neuron:138]: nit: These can be merged into a single `RUN` statement to reduce the number of layers in the image built from this dockerfile.

**Review by saimidu — COMMENTED** [review]
LGTM. Could you please also enable the `safety_check_test` on the developer config toml file, since this is a brand new dockerfile?

**Comment by saimidu** [comment]
There seems to have been a build failure caused by the renaming of the `neuron-start.sh` file.

**Review by saimidu — DISMISSED** [review]
Approved. Please revert config file changes.

**Review by saimidu — APPROVED** [review]
